src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -201,6 +224,8 @@ public class HubEndPoint<THub> : IInvocationBinder where THub : Hub
                 {
                     while (connection.Input.TryRead(out var buffer))
                     {
+                        buffer = connection.DataEncoder.Decode(buffer);
```

Comment:
This isn't right. This needs to be per decoded message inside of TryParseMessages